source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -231,17 +266,29 @@ bool lldb_private::formatters::swift::StringGuts_SummaryProvider(
         process->GetTarget().GetScratchClangASTContext()->GetBasicType(
             lldb::eBasicTypeObjCID);
 
-    // Warning: Using ValueObject::CreateValueObjectFromAddress can create an
-    // invalid ValueObject here in an unknown set of cases. Until this is fixed,
-    // use CreateValueObjectFromData (rdar://39741576).
+    bool success = false;
+    ValueObjectSP nsstring;
+
+    // Swift usually (but not always) wraps the NSString with one level of
+    // pointer indirection. So, first, dereference & format the string.
     DataExtractor DE(&startAddress, process->GetAddressByteSize(),
                      process->GetByteOrder(), process->GetAddressByteSize());
-    ValueObjectSP nsstring = ValueObject::CreateValueObjectFromData(
+    nsstring = ValueObject::CreateValueObjectFromData(
         "nsstring", DE, valobj.GetExecutionContextRef(), id_type);
-
     if (nsstring)
-      return NSStringSummaryProvider(*nsstring.get(), stream, summary_options);
-    return false;
+      success =
+          NSStringSummaryProvider(*nsstring.get(), stream, summary_options);
+
+    // We may have an NSString pointer inline, so try formatting it directly.
+    if (!success) {
```

Comment:
Is there really no way to tell whether this is boxed or inline?  Your workaround is okay; the chances that the first words of an inlined string are a valid pointer to another string being fairly small...  But it would be nice if the stdlib could provide a more definite way for us to figure this out.

Reply:
I've been staring at the bits inside of these strings and going through the scant set of docs, and can't find any indicator of whether the NSString is boxed. :/.